src/ngraph/op/fused/squeeze.cpp

```diff
@@ -38,25 +38,35 @@ void op::Squeeze::pre_validate_and_infer_types()
     auto data = input_value(0);
     auto axes_node = input_value(1).get_node_shared_ptr();
 
-    if (data.get_partial_shape().is_dynamic() || !axes_node->is_constant())
+    bool dynamic_data_rank = data.get_partial_shape().rank().is_dynamic();
+    bool dynamic_data_shape = data.get_partial_shape().is_dynamic();
+
+    bool axes_is_constant = axes_node->is_constant();
```

Comment:
Might be simpler if you said `axes_constant = as_type_ptr<axes_node>`. Then `axes_constant` will be null if it's not a constant, which might make the next lines simpler.

Reply:
Done